module/modmanager/manager_test.go

```diff
@@ -163,6 +163,31 @@ func TestModManagerFunctions(t *testing.T) {
 	test.That(t, err, test.ShouldNotBeNil)
 	test.That(t, err.Error(), test.ShouldContainSubstring, "no resource with name")
 
+	t.Log("test ReconfigureModule")
+	// Re-add counter1.
+	_, err = mgr.AddResource(ctx, cfgCounter1, nil)
+	test.That(t, err, test.ShouldBeNil)
+
+	// Change underlying binary path of module to be a copy of simplemodule/run.sh.
+	modCfg.ExePath = utils.ResolveFile("module/modmanager/data/simplemoduleruncopy.sh")
+
+	// Reconfigure module with new ExePath.
+	orphanedResourceNames, err := mgr.Reconfigure(ctx, modCfg)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, orphanedResourceNames, test.ShouldBeNil)
+
+	// counter1 should still be provided by reconfigured module.
+	ok = mgr.IsModularResource(rNameCounter1)
+	test.That(t, ok, test.ShouldBeTrue)
+
```

Comment:
Please add tests that the counter is still actually usable and working after reconfiguration, not just that it exists in the list. I'm definitely worried about this now that it's dealing more with orphans AND trying to close the old connection.